lib/redis/search-helper.js

```diff
@@ -3,11 +3,13 @@
 var PromiseHelper = require('./promise-helper')
 
 module.exports = class SearchHelper {
-  constructor(redisClient, searchString) {
+  constructor(redisClient, searchString, searchCriterion) {
```

Comment:
I'm thinking there's a cleaner way to do this than using `searchCriterion`.

Let's define a couple of constants on `Keys`:

```javascript
const SHORT_LINK_PREFIX = '/'
const TARGET_LINK_INDEX_PREFIX = 'target:'
```
Then we can replace `searchCriterion` with `keyPrefix` (which should be one of the above constants specified by the user), and replace `this.searchString` and `this.regExp` with the following:

```javascript
this.pattern = `${keyPrefix}*${searchString ? searchString + '*' : ''}`
```

Note this requires updating `this.regExp` to `this.pattern` in `processResults` as well. (Because it's not really a regular expression, but a glob-style pattern per: https://redis.io/commands/scan#the-match-option.)